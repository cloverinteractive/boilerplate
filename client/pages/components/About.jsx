import React from 'react';
import { Container, Header } from 'semantic-ui-react';
import { Helmet } from 'react-helmet';

const About = () => (
  <Container>
    <Helmet>
      <title>About</title>
    </Helmet>

    <Header as="h1">Made with love &lt;3!</Header>
    <p>
      We&apos;ve made this app because we spend too much time setting up new node apps.
    </p>

    <p>
      { "Think we've overlooked something? open up an " }
      <a href="https://github.com/cloverinteractive/boilerplate/issues">issue</a> in
      Github.
    </p>
  </Container>
);

export default About;
