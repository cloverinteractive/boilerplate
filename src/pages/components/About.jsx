import * as React from 'react';
import { Helmet } from 'react-helmet';

const About = () => (
  <div className="container is-fluid">
    <h1 className="title">Made with love &lt;3!</h1>
    <div className="content">
      <p>
        We&apos;ve made this boilerplate because we spend too much time setting up new node apps.
      </p>
      <p>
        Think we&apos;ve overlooked something? open up an
        {' '}
        <a href="https://github.com/cloverinteractive/boilerplate/issues">issue</a>
        {' '}
        in Github.
      </p>
    </div>
    <Helmet>
      <title>About</title>
    </Helmet>
  </div>
);

export default About;
