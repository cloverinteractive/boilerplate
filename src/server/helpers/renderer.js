import { renderToString } from 'react-dom/server';
import { Helmet } from 'react-helmet';
import serialize from 'serialize-javascript';
import { isDevelopment } from 'server/helpers/env-access';

export default (Component, store) => {
  const content = renderToString(Component);
  const helmet = Helmet.renderStatic();

  const productionStyles =
    `<link rel="stylesheet" href="/vendors.css">
    <link rel="stylesheet" href="/styles.css">`;

  const styles = isDevelopment ? '' : productionStyles;

  const template =
    `<!doctype html>
    <html lang="en-US">
      <head>
        <meta charset="utf-8">
        ${helmet.link.toString()}
        ${helmet.title.toString()}
        ${helmet.meta.toString()}
        ${styles}
        <script src="/vendors.js"></script>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.__INITIAL_STATE__ = ${serialize(store.getState())};
        </script>
        <script async src="/bundle.js"></script>
      </body>
    </html>`;

  return template;
};
