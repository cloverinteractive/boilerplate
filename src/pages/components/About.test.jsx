import * as React from 'react';
import renderer from 'react-test-renderer';
import About from './About';

describe('<About />', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(<About />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
