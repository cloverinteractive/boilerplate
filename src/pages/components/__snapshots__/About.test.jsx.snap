// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> it renders correctly 1`] = `
<div
  className="container is-fluid"
>
  <h1
    className="title"
  >
    Made with love &lt;3!
  </h1>
  <div
    className="content"
  >
    <p>
      We've made this boilerplate because we spend too much time setting up new node apps.
    </p>
    <p>
      Think we've overlooked something? open up an
       
      <a
        href="https://github.com/cloverinteractive/boilerplate/issues"
      >
        issue
      </a>
       
      in Github.
    </p>
  </div>
</div>
`;
